scripts/vim-patch.sh

```diff
@@ -177,19 +178,28 @@ list_vim_patches() {
   echo "  To port one of the above patches to Neovim, execute"
   echo "  this script with the patch revision as argument."
   echo
-  echo "  Examples: './scripts/vim-patch.sh 7.4.487'"
-  echo "            './scripts/vim-patch.sh 1e8ebf870720e7b671f98f22d653009826304c4f'"
+  echo "  Examples: '${BASENAME} 7.4.487'"
+  echo "            '${BASENAME} 1e8ebf870720e7b671f98f22d653009826304c4f'"
 }
 
-if [[ ${#} != 1 || "${1}" == "--help" || "${1}" == "-h" ]]; then
-  usage
+if [[ ${1} == "--help" || ${1} == "-h" ]]; then
+  usage 1
+  exit 0
+elif [[ ${#} != 1 ]]; then
+  usage 2
+  exit 1
+fi
+
+# Check if mercurial is in the user's PATH, and is executable
+if [[ ! -x $(command -v hg) ]]; then
```

Comment:
I read in `man bash` something along the lines of "variables, substring expansion ... disabled inside `[[...]]` blocks", is that not true? Shellcheck didn't warn about it.

On March 10, 2015 3:15:23 PM EDT, Florian Walch <notifications@github.com> wrote:
>>  }
>>  
>> -if [[ ${#} != 1 || "${1}" == "--help" || "${1}" == "-h" ]]; then
>> -  usage
>> +if [[ ${1} == "--help" || ${1} == "-h" ]]; then
>> +  usage 1
>> +  exit 0
>> +elif [[ ${#} != 1 ]]; then
>> +  usage 2
>> +  exit 1
>> +fi
>> +
>> +# Check if mercurial is in the user's PATH, and is executable
>> +if [[ ! -x $(command -v hg) ]]; then
>
>Add quotes around "$(..)"?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/neovim/neovim/pull/2108/files#r26155597